ui/src/openReadOnlyContent.ts

```diff
@@ -29,33 +29,71 @@ export async function openReadOnlyJson(node: { label: string, fullId: string },
     await openReadOnlyContent(node, content, '.json');
 }
 
-export async function openReadOnlyContent(node: { label: string, fullId: string }, content: string, fileExtension: string): Promise<void> {
+export async function openReadOnlyContent(node: { label: string, fullId: string }, content: string, fileExtension: string, options?: TextDocumentShowOptions): Promise<ReadOnlyContent> {
     if (!contentProvider) {
         contentProvider = new ReadOnlyContentProvider();
         ext.context.subscriptions.push(workspace.registerTextDocumentContentProvider(scheme, contentProvider));
     }
 
-    await contentProvider.openReadOnlyContent(node, content, fileExtension);
+    return await contentProvider.openReadOnlyContent(node, content, fileExtension, options);
+}
+
+export class ReadOnlyContent {
+    private _uri: Uri;
+    private _emitter: EventEmitter<Uri>;
+    private _content: string;
+
+    constructor(uri: Uri, emitter: EventEmitter<Uri>, content: string) {
+        this._uri = uri;
+        this._emitter = emitter;
+        this._content = content;
+    }
+
+    public get content(): string {
+        return this._content;
+    }
+
+    public async append(content: string): Promise<void> {
+        this._content += content;
+        this._emitter.fire(this._uri);
+    }
+
+    public clear(): void {
+        this._content = '';
+        this._emitter.fire(this._uri);
+    }
+
+    public async isVisible(): Promise<boolean> {
+        const visibleDocuments: TextDocument[] = window.visibleTextEditors.map(editor => editor.document);
+        return visibleDocuments.includes(await workspace.openTextDocument(this._uri));
```

Comment:
Isn't there a uri property on `editor.document` you could check? I think that would be more reliable as you're currently relying on object equality, which isn't the best. VS Code might make different instances of the same document class under the covers